pkg/results/utils.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package results
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// JSONFetch makes a GET request to the URL and marshals
+// the response into output.
+func JSONFetch(url string, output interface{}) error {
+	resp, err := http.Get(url) // #nosec
+	if err != nil {
+		return fmt.Errorf("%w: unable to fetch GET %s", err, url)
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("%w: unable to read body", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("received %d status with body %s", resp.StatusCode, body)
+	}
+
+	if err := json.Unmarshal(body, output); err != nil {
+		return fmt.Errorf("%w: unable to unmarshal JSON", err)
+	}
+
+	return nil
+}
```

Comment:
I usually do following instead but I guess it would be hard to return the body when it comes to error
```suggestion
	json.NewDecoder(resp.Body).Decode(v)
}
```

Reply:
I think it works well if the body is always JSON, however, I’m returning the error as raw text if something goes wrong (which means we won't be able to read it with the proposed approach). I'm going to keep my code for now.